File: app/src/main/java/com/mnj/java_solutions/linkedlist/RemoveNthNodeFromEndofList.java
package com.mnj.java_solutions.linkedlist;

public class RemoveNthNodeFromEndofList {

    public void removeNthFromEnd() {

        ListNode node = new ListNode();
        ListNode.Node head = node.insert();

        int n = 2;

        ListNode.Node temp = head;
        ListNode.Node fast = head;
        ListNode.Node slow = head;

        for (int i = 1; i <= n + 1; i++) {
            fast = fast.next;
        }

        while (fast != null) {
            slow = slow.next;
            fast = fast.next;
        }

        slow.next = slow.next.next;

        node.display(head);
    }
}

File: app/src/main/java/com/mnj/java_solutions/linkedlist/RemoveNthNodeFromEndofList.kt
package com.mnj.java_solutions.linkedlist

import com.mnj.java_solutions.linkedlist.ListNode

class RemoveNthNodeFromEndofList {
    fun removeNthFromEnd() {
        val node = ListNode()
        val head = node.insert()
        val n = 2
        val temp = head
        var fast = head
        var slow = head
        for (i in 1..n + 1) {
            fast = fast!!.next
        }
        while (fast != null) {
            slow = slow.next
            fast = fast.next
        }
        slow.next = slow.next.next
        node.display(head)
    }
}